### ISSUE TITLE
Failed to build from source with mpack 1.0.10-0

### ISSUE THREAD

**Issue by uraza** [neovim/neovim#23022]
### Problem

Hello.

I tried building Neovim 0.9.0 from sources but ran into an error.

After "git checkout stable", the build fails:

```
[ 53%] Generating usr/lib/luarocks/rocks-5.1/mpack
Warning: Failed searching manifest: Failed downloading https://luarocks.org/manifest-5.1 - failed downloading https://luarocks.org/manifest-5.1

Error: Could not find a result named mpack 1.0.10-0: No results matching query were found for Lua 5.1.
To check if it is available for other Lua versions, use --check-lua-versions.
```

If I revert back to mpack 1.0.8-0 (the one that was used in Neovim 0.8.2 for example):

```
sed -i 's/1.0.10-0/1.0.8-0/g' ./cmake.deps/cmake/BuildLuarocks.cmake
sed -i 's/1.0.10-0/1.0.8-0/g' ./.deps/CMakeFiles/mpack.dir/build.make
```

Then the build passes:

```
[ 53%] Generating usr/lib/luarocks/rocks-5.1/mpack
Warning: Failed searching manifest: Failed downloading https://luarocks.org/manifest-5.1 - failed downloading https://luarocks.org/manifest-5.1

/XXX/gcc -O2 -fPIC -I/YYY/neovim/.deps/usr/include/luajit-2.1 -c lmpack.c -o lmpack.o
/XXX/gcc  -shared -o mpack.so lmpack.o
mpack 1.0.8-0 is now installed in /YYY/neovim/.deps/usr (license: MIT)

make[3]: Leaving directory `/ZZZ/neovim/.deps'
[ 53%] Built target mpack
```

Any idea on how I could fix this and use the newer mpack version?

Thanks.

### Steps to reproduce

git clone https://github.com/neovim/neovim
cd neovim/
git checkout stable
make CMAKE_BUILD_TYPE=Release CMAKE_EXTRA_FLAGS="-DCMAKE_INSTALL_PREFIX=$HOME/shared/neovim/neovim-0.9.0"

### Expected behavior

Passing build with mpack 1.0.10-0.

### Neovim version (nvim -v)

NVIM v0.9.0

### Vim (not Nvim) behaves the same?

Not applicable

### Operating system/version

CentOS Linux 7

### Terminal name/version

GNOME Terminal 3.28.2

### $TERM environment variable

xterm

### Installation

Build from repo

**Comment by clason** [comment]
Try again; this is luarocks being flaky.

**Comment by clason** [comment]
Are you behind a firewall or using a proxy?

**Comment by uraza** [comment]
Hum, most likely (corporate virtual machine, who knows what can happen there? :) ).
The troubling part is that the earlier version work.

**Comment by clason** [comment]
I suspect some issues with luarocks CDN not mirroring the new version properly. We get transient failures on CI as well.

**Comment by uraza** [comment]
Yes, that is possible. Hopefully it gets sorted out sooner or later.
In the meantime, I have been able to compile Neovim by reverting back to mpack 1.0.8-0 (using the sed commands from the issue description) and I have not observed any issue when using it so far.
Thanks!

**Comment by dundargoc** [comment]
@uraza can you try installing wget and try again? I managed to work around the same problem on docker alpine with this.
